src/validators.ts

```diff
@@ -45,7 +46,18 @@ function inParallel<I, O, E>(
 ): Validator<I, O, E> {
   return (input) =>
     Promise.all(validators.map((validator) => validator(input))).then(
-      (results) => results[0]
+      (results) => {
+        const errors = results.flatMap((r) =>
+          matchResult(r, {
+            success: () => [] as E[],
+            failure: (failure) => failure,
+          })
+        );
+
+        return errors.length > 0
+          ? failure(errors as NonEmptyArray<E>)
+          : results[0];
```

Comment:
I'm not sure how to handle the success case when validators are parallel

Reply:
I'm not even sure it compiles, to be honest, the general idea of inParallel is they validator a should return the same type and the runtime value could be any of the returned ones.

Reply:
Sorry for stepping in, but given that the returning type of each validator is the same would it make sense at API level to let the user _optionally_ decide which validator output consider as overall returning value? I'm thinking something like a semigroup to reduce the multiple outputs to a single one, I can't think a real use case scenario though